lib/src/proxy-authorization.class.ts

```diff
@@ -0,0 +1,49 @@
+import { BaseHeader } from "./base-header.class";
+
+export class ProxyAuthorization extends BaseHeader {
+  protected _type?: string | undefined;
+  protected _credentials?: string | undefined;
+
+  getType(): string | undefined {
+    return this._type;
+  }
+
+  setType(value: string | undefined) {
```

Comment:
Similar to the Proxy-Authentication header, we can use a "Authentication Schema" type here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes